src/libLookup/Lookup.cpp

```diff
@@ -1642,17 +1708,417 @@ bool Lookup::InitMining()
+
+            LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+                      "IP:" << peer.GetPrintableIPAddress());
+        }
+    }
+
+    P2PComm::GetInstance().SendMessage(requestingNode, offlineLookupsMessage);
 #endif // IS_LOOKUP_NODE
+    return true;
+}
+
+bool Lookup::ProcessSetOfflineLookups(const std::vector<unsigned char>& message,
+                                      unsigned int offset, const Peer& from)
+{
+    // Message = [num_offline_lookups][LookupPeer][LookupPeer]... num_offline_lookups times
+    LOG_MARKER();
+#ifndef IS_LOOKUP_NODE
+    if (IsMessageSizeInappropriate(message.size(), offset, sizeof(uint32_t)))
+    {
+        return false;
+    }
+
+    uint32_t numOfflineLookups
+        = Serializable::GetNumber<uint32_t>(message, offset, sizeof(uint32_t));
+    offset += sizeof(uint32_t);
+
+    LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "ProcessSetOfflineLookups sent by "
+                  << from << " for numOfflineLookups " << numOfflineLookups);
+
+    if (IsMessageSizeInappropriate(message.size(), offset,
+                                   (IP_SIZE + PORT_SIZE) * numOfflineLookups))
+    {
+        return false;
+    }
+
+    for (unsigned int i = 0; i < numOfflineLookups; i++)
+    {
+        Peer peer(message, offset);
+        offset += (IP_SIZE + PORT_SIZE);
+
+        // Remove selfPeerInfo from m_lookupNodes
+        auto iter = std::find(m_lookupNodes.begin(), m_lookupNodes.end(), peer);
+        if (iter != m_lookupNodes.end())
+        {
+            m_lookupNodesOffline.push_back(*iter);
+            m_lookupNodes.erase(iter);
+
+            LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+                      "ProcessSetOfflineLookups recvd offline lookup "
+                          << i << ": " << peer);
+        }
     }
 
+    {
+        unique_lock<mutex> lock(m_mutexOfflineLookupsUpdation);
+        m_fetchedOfflineLookups = true;
+        m_offlineLookupsCondition.notify_one();
+    }
+#endif // IS_LOOKUP_NODE
+    return true;
+}
+
+#ifdef IS_LOOKUP_NODE
+void Lookup::StartSynchronization()
+{
+    LOG_MARKER();
+
+    this->CleanVariables();
+
+    auto func = [this]() -> void {
+        GetMyLookupOffline();
+        GetDSInfoFromLookupNodes();
+        while (m_syncType != SyncType::NO_SYNC)
+        {
+            GetDSBlockFromLookupNodes(m_mediator.m_dsBlockChain.GetBlockCount(),
+                                      0);
+            GetTxBlockFromLookupNodes(m_mediator.m_txBlockChain.GetBlockCount(),
+                                      0);
+            this_thread::sleep_for(chrono::seconds(NEW_NODE_SYNC_INTERVAL));
+        }
+    };
+    DetachedFunction(1, func);
+}
+
+Peer Lookup::GetLookupPeerToRsync()
+{
+    LOG_MARKER();
+
+    std::vector<Peer> t_Peers;
+    for (auto p : m_lookupNodes)
+    {
+        if (p == m_mediator.m_selfPeer)
+        {
+            // SKIP
```

Comment:
Rather than //skip, can we put "continue" or refine the logic

if (p == m_mediator.m_selfPeer)
{
t_Peers.push_back(p);
}

Reply:
ok

Reply:
Not yet addressed